### ISSUE TITLE
[1.0.0.18][Minor] Milla's cubes have incorrect collision on the inside of all four large sloped tiles

### ISSUE THREAD

**Issue by benjistring** [whitelilydragon/ShangMuArchitect#143]
https://user-images.githubusercontent.com/69790793/108471125-cfd0b600-723f-11eb-9b89-434f255f05a8.mp4

**Comment by whitelilydragon** [comment]
I'm sorry, but I have to label this as #Won't Fix, because of how the cube works:

A) The player entity is already a fake object projected by a logic controller in the corner of the room, pretending to interact with tiles based on collision maps stored in its allocated memory. The cube is an extension of this, being a fake object thrown by a fake player onto fake terrain! It's very abstract nonsense I have developed a contempt and regret for, don't you worry about that. ; ^ ; Do not architect game engines this way!

B) Because Shang Mu Architect's collision maps are 1-dimension, instead of 2-dimensional, the engine has to assume how to interact with floors, walls, and ceilings differently, depending on the player's angle. (For instance, assuming all data below a certain point on the graph is solid ground, or all data points to the right of collision points are all solid walls.) It is difficult to extend this assumption system to the cube, as given in Point A.

C) As a consequence of collision maps being 1D, double-sided slope/ramp tiles need to swap out their collision maps based on the player's current position within the immediate area of the tile. (Swapping out data 24-32 pixels ahead of the player's position, to avoid collision warping!)

D) Point C involves switching out the tiles themselves, which could break the player's own collisions. For instance, if the cube wants to fall onto the outside of a slope tile, while the player is running along the inside of it, they would either warp to the outside of the slope, or get crushed and die! Enemies avoid this by assuming which versions of inside/outside tile data they should call, instead of swapping out the actual tiles themselves, but this still wouldn't be an ideal system for the cube; it would fall some distance into some slopes before colliding with them.

Basically, the cube breaks the laws of this engine's physics! If I ever decide to refactor the game's collision map system to become properly 2D masks, this could be fixed indirectly, because of how those masks could work, but as things stand now- it would be an additional strain on CPU time and scope to allow the cube to interact with more slope tiles as you'd expect.

I'm sorry ; ^ ;
